training_engine_sae.py

```diff
@@ -0,0 +1,152 @@
+import cv2
+import numpy as np
+import random as rd
+from keras.models import Model
+from keras.layers import Dropout, UpSampling2D, Concatenate
+from keras.layers import Conv2D, MaxPooling2D, Input
+from keras.optimizers import Adam
+from keras.callbacks import EarlyStopping,ModelCheckpoint
+from keras.backend import image_data_format
+
+# ===========================
+#       SETTINGS
+# ===========================
+
+VALIDATION_SPLIT=0.2
+BATCH_SIZE = 16
+
+# ===========================
+
+def get_input_shape(height, width, channels = 3):
+    if image_data_format() == 'channels_first':
+        return (channels, height, width)
+    else:
+        return (height, width, channels)
+
+
+def get_sae(height, width, pretrained_weights = None):
+    ff = 32
```

Comment:
Similar question with the integer for the number of output filters in the convolution.

Reply:
The network model can be tuned in many different ways so I don't see the point in leaving one hyper-parameter to be specified by the user.